packages/cascara/src/modules/DataCheckbox/DataCheckbox.js

```diff
@@ -31,26 +31,39 @@ const DataCheckbox = ({
   const { isEditing, formMethods } = useContext(ModuleContext);
   const checkbox = useCheckboxState({ state: Boolean(value) });
 
+  // do not add aria label if tag is used
+  const ariaLabel = isLabeled ? {} : { 'aria-label': label };
+
   const renderEditing = (
-    <label htmlFor={label}>
+    <>
+      {label && isLabeled && (
+        <label htmlFor={attribute || label}>
+          <span className={styles.LabelText}>{label || attribute}</span>
+        </label>
+      )}
       <Checkbox
+        {...ariaLabel}
         {...rest}
         {...checkbox}
         className={styles.Input}
-        id={label}
+        id={attribute || label}
```

Comment:
```suggestion

```
I think we can remove this one since we would be conditionally spreading it above from the ariaLabel const